sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/CatalystTypeConverters.scala

```diff
@@ -284,9 +321,9 @@ object CatalystTypeConverters {
       row: Row,
       schema: StructType,
       converters: Array[Any => Any]): Row = {
-    val ar = new Array[Any](row.size)
+    val ar = new Array[Any](converters.size)
     var idx = 0
-    while (idx < row.size) {
+    while (idx < converters.size && idx < row.size) {
```

Comment:
For HiveQuerySuite "Add JAR command 2", the size of row is 1, but the number of fields is 0, maybe Hive 1.3 has changed the result?